lib/modules/manager/dockerfile/extract.ts

```diff
@@ -154,13 +154,31 @@ const quayRegex = regEx(/^quay\.io(?::[1-9][0-9]{0,4})?/i);
 
 export function getDep(
   currentFrom: string | null | undefined,
-  specifyReplaceString = true
+  specifyReplaceString = true,
+  registryAliases?: Record<string, string>
 ): PackageDependency {
   if (!is.string(currentFrom)) {
     return {
       skipReason: 'invalid-value',
     };
   }
+
+  // Resolve registry aliases first so that we don't need special casing later on:
+  for (const [name, value] of Object.entries(registryAliases ?? {})) {
```

Comment:
please add tests to dockerfile manager to get full coverage for this manager when tested alone. you only need to test the getDep function

Reply:
Sure, done. Produces a little bit duplication in the tests, but I guess that's fine 😄